### PR TITLE
Initial support for github actions

### PR DESCRIPTION
**Pull request by sbc100**
This adds a basic workflow that builds and tests wabt on all three
desktop platforms.
    
The plan is to extend this to completely replace travis and appveyor
in the future.

Fixes: #1168

### CODE REVIEW

File: test/run-tests.py [hunk 1]
```diff
@@ -902,8 +903,8 @@ def main(args):
     variables = {}
     variables['test_dir'] = os.path.abspath(TEST_DIR)
     variables['bindir'] = options.bindir
-    variables['gen_wasm_py'] = find_exe.GEN_WASM_PY
-    variables['gen_spec_js_py'] = find_exe.GEN_SPEC_JS_PY
+    variables['gen_wasm_py'] = os.path.join(TEST_DIR, 'gen-wasm.py')
```
Comment (dschuff) [inline test/run-tests.py]: do these refactorings have anything to do with the CI part?

Comment (sbc100) [reply test/run-tests.py]: I'm gonna split them out.

**Comment by sbc100** [comment]
OK this should be good to go now.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, macos-latest, windows-latest]
+    steps:
+    - uses: actions/setup-python@v1
+      with:
+        # Using 2.8 results in cmake finding the 2.7 on the windows builder
+        # at C:/ProgramData/chocolatey/bin/python2.7.exe and this version seems
+        # to have some issues.
+        python-version: '2.7'
```
Comment (dschuff) [inline .github/workflows/build.yml]: no py3 available?

Comment (sbc100) [reply .github/workflows/build.yml]: wabt tests are currently broken under python3: https://github.com/WebAssembly/wabt/issues/1248

We need to fix that but thats a separate issue.

**Pull request merged**